SimPEG/EM/Static/IP/ProblemIP_2D.py

```diff
@@ -0,0 +1,425 @@
+            A = self.getA(ky)
+            self.Ainv[iky] = self.Solver(A, **self.solverOpts)
+            RHS = self.getRHS(ky)
+            u = self.Ainv[iky] * RHS
+            self.f[Srcs, self._solutionType, iky] = u
+
+    def fields(self, m):
+        # This is stupid..., but not sure how I can change InvProblem
+        # where calling self.fields
+        return None
+
+    def getJ(self, f=None):
+        """
+            Generate Full sensitivity matrix
+        """
+
+        if self.verbose:
+            print (">> Compute Sensitivity matrix")
+
+        if self.f is None:
+            self.fieldsdc()
+
+        Jt = []
+
+        # Assume y=0.
+        # This needs some thoughts to implement in general when src is dipole
+        dky = np.diff(self.kys)
+        dky = np.r_[dky[0], dky]
+        y = 0.
+        for src in self.survey.srcList:
+            for rx in src.rxList:
+                Jtv_temp1 = np.zeros((self.model.size, rx.nD), dtype=float)
+                Jtv_temp0 = np.zeros((self.model.size, rx.nD), dtype=float)
+                Jtv = np.zeros((self.model.size, rx.nD), dtype=float)
+                # TODO: this loop is pretty slow .. (Parellize)
+                for iky in range(self.nky):
+                    u_src = self.f[src, self._solutionType, iky]
+                    ky = self.kys[iky]
+                    AT = self.getA(ky)
+
+                    # wrt f, need possibility wrt m
+                    P = rx.getP(self.mesh, rx.projGLoc(self.f)).toarray()
+
+                    ATinvdf_duT = self.Ainv[iky] * (P.T)
+
+                    dA_dmT = self.getADeriv(ky, u_src, ATinvdf_duT,
+                                            adjoint=True)
+                    Jtv_temp1 = 1./np.pi*(-dA_dmT)
+                    if rx.nD == 1:
+                        Jtv_temp1 = Jtv_temp1.reshape([-1, 1])
+
+                    # Trapezoidal intergration
+                    if iky == 0:
+                        # First assigment
+                        Jtv += Jtv_temp1*dky[iky]*np.cos(ky*y)
+                    else:
+                        Jtv += Jtv_temp1*dky[iky]/2.*np.cos(ky*y)
+                        Jtv += Jtv_temp0*dky[iky]/2.*np.cos(ky*y)
+                    Jtv_temp0 = Jtv_temp1.copy()
+                Jt.append(Jtv)
+
+        return np.hstack(Jt).T
+
+    def Jvec(self, m, v, f=None):
+        self.model = m
+        if self.fswitch == False:
+            f = self.fieldsdc()
+            self.J = self.getJ(f=f)
+            self.fswitch = True
+        return self.sign * self.J.dot(v)
+
+    def Jtvec(self, m, v, f=None):
+        self.model = m
+        if self.fswitch == False:
+            f = self.fieldsdc()
+            self.J = self.getJ(f=f)
+            self.fswitch = True
+        Jtvec = self.J.T.dot(v)
+        return self.sign * Jtvec
+
+    @property
+    def deleteTheseOnModelUpdate(self):
+        toDelete = []
+        return toDelete
+
+    # assume log rho or log cond
+    @property
```

Comment:
do these need to be repeated? or can we just inherit from baseEM?

Reply:
there are functions that I am using from BaseDCProblem_2D. 
for IP we do not need to delete mass matricies.